crates/noirc_frontend/src/hir/scope/mod.rs

```diff
@@ -26,7 +26,7 @@ pub struct Scope<K, V>(pub HashMap<K, V>);
 
 impl<K: std::hash::Hash + Eq + Clone, V> Scope<K, V> {
     pub fn new() -> Self {
-        Scope(HashMap::with_capacity(100))
+        Scope(HashMap::with_capacity(10))
```

Comment:
I lowered it because 10 seems quite high for a scope. Every function body is a new scope, every for loop body, every if-then block, every if-else block etc. Its very rare I see 10 or more variables defined in one block. Imports are handled separately in an ItemScope so they don't add to this number.

Reply:
I guess I just don't see the benefit in lowering it. Even this very basic program (https://github.com/vezenovm/simple_shield/blob/master/circuits/src/main.nr) gets close to 10 in the main function block. I feel like it definitely uncommon to see more than 10 variables in one block but I don't know if it will be rare, especially while a program is still in development and more advanced programs are created with Noir.

Reply:
with_capacity will create a HashMap with 10 as its initial capacity, not maximum capacity. So by saying 10 here, we're saying 10 is the minimum we expect and any block that uses less than that is using extra memory. In the case a scope uses more than 10, it reallocates.

Reply:
ah, ok thank you for that clarification. I should've looked up exactly what with_capacity did